tests/system/python/api/test_service.py

```diff
@@ -0,0 +1,162 @@
+        core_svc = jdoc['services'][1]
+        assert isinstance(core_svc['management_port'], int)
+        assert 8081 == core_svc['service_port']
+        assert 'running' == core_svc['status']
+        assert 'Core' == core_svc['type']
+        assert '0.0.0.0' == core_svc['address']
+        assert 'FogLAMP Core' == core_svc['name']
+        assert 'http' == core_svc['protocol']
+
+    @pytest.mark.parametrize("plugin, svc_name, config, enabled, svc_count", [
+        ("randomwalk", SVC_NAME_1, None, True, 3),
+        ("http_south", SVC_NAME_2, None, False, 3),
+        ("Benchmark", SVC_NAME_3, None, True, 4)
+    ])
+    def test_add_service(self, foglamp_url, wait_time, plugin, svc_name, config, enabled, svc_count):
+        jdoc = plugin_and_service.add_south(plugin, foglamp_url, svc_name, config, enabled)
+        assert svc_name == jdoc['name']
+        assert UUID(jdoc['id'], version=4)
+
+        time.sleep(wait_time)
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/service')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert len(jdoc), "No data found"
+        assert svc_count == len(jdoc['services'])
+        southbound_svc = jdoc['services'][svc_count - 1]
+        assert isinstance(southbound_svc['management_port'], int)
+        assert southbound_svc['service_port'] is None
+        assert svc_name if enabled else SVC_NAME_1 == southbound_svc['name']
+        assert 'running' == southbound_svc['status']
+        assert 'Southbound' == southbound_svc['type']
+        assert 'localhost' == southbound_svc['address']
+        assert 'http' == southbound_svc['protocol']
+
+    def test_add_service_with_config(self, foglamp_url):
+        data = {"name": SVC_NAME_4,
+                "type": "South",
+                "plugin": 'randomwalk',
+                "config": {"maxValue": {"value": "20"}, "assetName": {"value": "Random"}},
+                "enabled": True
+                }
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("POST", '/foglamp/service', json.dumps(data))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert SVC_NAME_4 == jdoc['name']
+        assert UUID(jdoc['id'], version=4)
+
+        conn.request("GET", '/foglamp/category/{}'.format(SVC_NAME_4))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert len(jdoc), "No data found"
+        assert data['config']['assetName']['value'] == jdoc['assetName']['value']
+        assert data['config']['maxValue']['value'] == jdoc['maxValue']['value']
+
+    @pytest.mark.parametrize("encoded_svc_name, svc_name, status, svc_count", [
+        ("FogLAMP%20Storage", "FogLAMP Storage", 404, 2),
+        ("FogLAMP%20Core", "FogLAMP Core", 404, 2),
```

Comment:
should be a BAD REQUEST or NOT ALLOWED!

Reply:
https://github.com/foglamp/FogLAMP/blob/050dbf03f3a9eed5c37cf7acc5b97529f810bbba/python/foglamp/services/core/api/service.py#L87
there is no other way to find the service existence; we made it from schedule entry and return HTTP code status is 404 if not found

Otherwise we need to place new check (special case) for FogLAMP Storage and FogLAMP Core service

Reply:
Okay, we should file an issue (to place the check) with low priority and add a TODO here.